test/e2e/flowaggregator_test.go

```diff
@@ -259,18 +266,49 @@ func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP stri
 						t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
 					}
 				}
-				// Check the bandwidth using octetDeltaCount in data record.
+				curOctetTotalCount := getUnit64FieldFromRecord(t, record, "octetTotalCountFromSourceNode")
+				if curOctetTotalCount > octetTotalCount {
+					octetTotalCount = curOctetTotalCount
+				}
 				if checkBandwidth {
-					checkBandwidthFromRecord(t, record, bandwidth)
+					exportTime := int64(getUnit64FieldFromRecord(t, record, "flowEndSeconds"))
+					if exportTime <= timeStartSec+iperfTime-aggregatorExportInterval && exportTime > timeStartSec+aggregatorExportInterval {
+						curOctetDeltaCount := getUnit64FieldFromRecord(t, record, "octetDeltaCountFromSourceNode")
+						t.Logf("Check the bandwidth using octetDeltaCountFromSourceNode in data record.")
+						checkBandwidthByInterval(t, bandwidth, curOctetDeltaCount, float64(aggregatorExportInterval))
+					}
 				}
 			}
 		}
 	}
+	if checkBandwidth && octetTotalCount > 0 {
+		t.Logf("Check the average bandwidth using octetTotalCountFromSourceNode in data record.")
+		checkBandwidthByInterval(t, bandwidth, octetTotalCount, float64(iperfTime))
+	}
 	// Checking only data records as data records cannot be decoded without template
 	// record.
 	assert.GreaterOrEqualf(t, dataRecordsCount, expectedNumDataRecords, "IPFIX collector should receive expected number of flow records. Considered records: ", len(recordSlices))
 }
 
+func checkBandwidthByInterval(t *testing.T, bandwidth string, octetCount uint64, interval float64) {
+	// bandwidth from iperf output
+	bwSlice := strings.Split(bandwidth, " ")
+	iperfBandwidth, err := strconv.ParseFloat(bwSlice[0], 64)
+	require.NoError(t, err, "Error when converting iperf bandwidth to float64 type")
+
+	recBandwidth := float64(octetCount) * 8 / 1024 / 1024 / interval
+	var recBandwidthStr string
+	if strings.Contains(bwSlice[1], "Mbits") {
+		recBandwidthStr = fmt.Sprintf("%.2f Mbits/sec", recBandwidth)
+	} else {
+		// iperfBandwidth is in Gbits/sec
+		iperfBandwidth = iperfBandwidth * float64(1024)
+		recBandwidthStr = fmt.Sprintf("%.2f Gbits/sec", recBandwidth/1024)
+	}
+	t.Logf("Iperf throughput: %s, IPFIX record throughput: %s", bandwidth, recBandwidthStr)
+	assert.InDeltaf(t, recBandwidth, iperfBandwidth, iperfBandwidth*0.15, "Difference between Iperf bandwidth and IPFIX record bandwidth should be lower than 15%")
```

Comment:
This will be checked for both octetDelta and octetTotal fields right? Can the message differentiate the error message for different fields?

Reply:
Yes, add a field argument to differentiate error messages.